### ISSUE TITLE
Delve hangs

### ISSUE THREAD

**Issue by nuliknol** [derekparker/delve#1145]
1. What version of Delve are you using (`dlv version`)?
Delve Debugger
Version: 1.0.0-rc.2
Build: $Id: 5c7676404d10d52e16be57fea744a7312b831f0c $

2. What version of Go are you using? (`go version`)?
go version go1.10rc2 linux/amd64

3. What operating system and processor architecture are you using?
Ubuntu 16.04

4. What did you do?
I executed my code with delve with the purpose of debugging it

5. What did you expect to see?
I expected delve to debug the code

6. What did you see instead?
After few `step`s delve entered infinite loop, which does not happen in the program when running without delve. Interrupting it showed this message:

received SIGINT, stopping process (will not forward signal)> runtime.futex() /usr/local/go/src/runtime/sys_linux_amd64.s:527 (PC: 0x462723)
Warning: debugging optimized function
	breakpoint hit during next, continuing...
received SIGINT, stopping process (will not forward signal)> runtime.futex() /usr/local/go/src/runtime/sys_linux_amd64.s:527 (PC: 0x462723)
Warning: debugging optimized function
	breakpoint hit during next, continuing...
received SIGINT, stopping process (will not forward signal)> runtime.futex() /usr/local/go/src/runtime/sys_linux_amd64.s:527 (PC: 0x462723)
Warning: debugging optimized function
	breakpoint hit during next, continuing...

So, basically, it just hangs in the middle of the debug session and it happens randomly.

**Comment by derekparker** [comment]
Do you have an example of the code you were trying to debug or any way to reproduce this?
